File: from-2501-to-2600/2583-kth-largest-sum-in-a-binary-tree/solution.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
 func kthLargestLevelSum(root *TreeNode, k int) int64 {
    q := []*TreeNode{root}
    sums := []int64{}
    var lvlSum int64

    for len(q) > 0 {
        n := len(q)
        lvlSum = 0
        for i := 0; i < n; i ++ {
            node := q[0]
            q = q[1:]
            lvlSum += int64(node.Val)
            if node.Left != nil {
                q = append(q, node.Left)
            }
            if node.Right != nil {
                q = append(q, node.Right)
            }
        }
        sums = append(sums, lvlSum)
    }
    if len(sums) < k {
        return -1
    }
    sort.Slice(sums, func(i, j int) bool{
        return sums[i] > sums[j]
    })
    return sums[k - 1]
}

File: from-2501-to-2600/2583-kth-largest-sum-in-a-binary-tree/solution.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def kthLargestLevelSum(self, root: Optional[TreeNode], k: int) -> int:
        q = deque([root])
        sums = []
        while q:
            n = len(q)
            levelSum = 0
            for i in range(n):
                node = q.popleft()
                levelSum += node.val
                if node.left:
                    q.append(node.left)
                if node.right:
                    q.append(node.right)
            sums.append(levelSum)
        if len(sums) < k:
            return -1

        sums.sort(reverse=True)
        return sums[k-1]
            